deck-of-cards-kata/src/test/java/bnymellon/codekatas/deckofcards/APIComparisonCalculator.java

```diff
@@ -0,0 +1,101 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package bnymellon.codekatas.deckofcards;
+
+import org.eclipse.collections.api.RichIterable;
+import org.eclipse.collections.api.set.sorted.MutableSortedSet;
+import org.eclipse.collections.api.tuple.Pair;
+import org.eclipse.collections.api.tuple.Twin;
+import org.eclipse.collections.impl.factory.SortedSets;
+import org.eclipse.collections.impl.tuple.Tuples;
+import org.eclipse.collections.impl.utility.ArrayIterate;
+import org.eclipse.collections.impl.utility.StringIterate;
+
+import java.lang.reflect.Method;
+
+public class APIComparisonCalculator {
+
+        public Twin<MutableSortedSet<String>> compare(Class<?> leftClass, Class<?> rightClass)
+        {
+            Twin<Class<?>> classPair = Tuples.twin(leftClass, rightClass);
+            MutableSortedSet<String> leftMethods =
+                    this.getMethodNames(classPair.getOne());
+            MutableSortedSet<String> rightMethods =
+                    this.getMethodNames(classPair.getTwo());
+
+            MutableSortedSet<String> symmetricDifference = leftMethods.symmetricDifference(rightMethods);
+            this.output(classPair, "Symmetric Difference",
+                    symmetricDifference);
+            MutableSortedSet<String> intersection = leftMethods.intersect(rightMethods);
+            this.output(classPair, "Intersection",
+                    intersection);
+            return Tuples.twin(symmetricDifference, intersection);
+        }
+
+        private void output(Twin<Class<?>> classPair,
+                            String operation,
+                            RichIterable<String> strings)
+        {
+            this.outputTitle(operation + " (" + this.classNames(classPair) + ")");
+            this.outputGroupByToString(strings);
+        }
+
+        private String classNames(Twin<Class<?>> classPair)
+        {
+            return classPair.getOne().getSimpleName() +
+                    ", " +
+                    classPair.getTwo().getSimpleName();
+        }
+
+        private MutableSortedSet<String> getMethodNames(Class<?> classOne)
+        {
+            return ArrayIterate.collect(
+                    classOne.getMethods(),
+                    this::methodNamePlusParms,
+                    SortedSets.mutable.empty());
+        }
+
+        private String methodNamePlusParms(Method method)
+        {
+//            return method.getName() + "(" + this.parameterNames(method) + ")";
```

Comment:
Did you mean to keep this commented out line?

Reply:
Yes, good catch. It's useful to swap these if I want to see methods unique only by name or by parameter types as well. I could alternatively pass a boolean when I construct the class.

Reply:
Adding a boolean might be a better approach, to eliminate someone from accidentally removing that commented out line.

Reply:
Agreed. Fixing now. Thanks for the feedback!